### DISCUSSION TITLE
How to migrate filename_disk in Directus_files to v9?

### DISCUSSION THREAD

**Discussion by cdwmhcc** [directus/directus#4763] [Archive]
v8 `filename_disk` stores the file name + extension, while v9 `filename_disk` stores the UUID + extension. 
I tried to modify the filename_disk in the v8 old data to uuid + extension, but it didn't work.
How to migrate data?

![image](https://user-images.githubusercontent.com/8030134/113037760-4e6a2d00-91c8-11eb-9b06-d388778e1586.png)

**Comment by cdwmhcc** [comment]
I think it is unreasonable to store filename_disk as uuid + file extension in v9. The most important thing is that the actual name of the stored file will also become uuid, which will cause confusion in the display of data in s3 and other systems. Data synchronization causes trouble.

I usually upload a lot of files, this only represents my thoughts, maybe it's wrong.

**Reply by cdwmhcc** [reply]
I tried to modify the file name in s3 and the `filename_disk` in the database to uuid + extension, but it still does not display.

Did I miss something?

**Comment by cdwmhcc** [comment]
I found the reason, because I use uuid v5, but directus_files id must use uuid v4. When I use uuid v4, the files can be displayed normally in v9.

I tried migration-tool, but an error occurred during the conversion process. I don't know the specific reason, and the console did not give a specific error message.

Now how to manually convert the id in the old database to uuid v4? What are its rules?

**Comment by cdwmhcc** [comment]
I tried many times and came to the conclusion:

Even if I convert directus_files id to uuid v4, the names of tens of thousands of files uploaded to s3 via v8 need to be renamed to uuid v4, which is unimaginable!

Maybe I misunderstood or missed some details, please correct me.

**Comment by rijkvanzanten** [comment]
Heya! You don't have to migrate the filename_disk necessarily. Directus will use whatever's stored in the DB to read from the bucket 👍🏻 

You are correct in that the file primary key is now a UUIDv4 instead of an integer.

This works as expected:

<img width="1400" alt="CleanShot 2021-03-31 at 10 22 41@2x" src="https://user-images.githubusercontent.com/9141017/113159982-0e786980-920b-11eb-9076-13d90bea0731.png">

<img width="192" alt="CleanShot 2021-03-31 at 10 22 59@2x" src="https://user-images.githubusercontent.com/9141017/113160017-16d0a480-920b-11eb-80cb-b08903992ba5.png">

**Reply by cdwmhcc** [reply]
This is indeed the case, just change the id to uuid v4 and it will work.

In v8, I used Repeater for almost every item and added the files field. As you know, the data in Repeater is hard-coded. Using uuid v4 is undoubtedly disastrous for my data migration.

**Reply by cdwmhcc** [reply]
But I believe I can finally solve it, and I need to write a few more loop judgments.

**Reply by cdwmhcc** [reply]
This issue has been resolved.
